lib/matplotlib/cbook/__init__.py

```diff
@@ -248,14 +248,22 @@ class silent_list(list):
     one will get ::
 
         <a list of 3 Line2D objects>
+
+    If ``self.type`` is None, the type name is obtained from the first item in
+    the list (if any).
     """
+
     def __init__(self, type, seq=None):
         self.type = type
         if seq is not None:
             self.extend(seq)
 
     def __repr__(self):
-        return '<a list of %d %s objects>' % (len(self), self.type)
+        return (f"<a list of {len(self)} {self.type} objects>"
+                if self.type is not None else
```

Comment:
Can you pull out
~~~
type = self.type if self.type is not None else ...
~~~
That'd make the messages simpler to read.

Reply:
sure